server/testutil/testmetrics/testmetrics.go

```diff
@@ -67,6 +68,32 @@ func HistogramVecValues(t testing.TB, metric *prometheus.HistogramVec) []Histogr
 	return out
 }
 
+// AssertHistogramSamples asserts that the given histogram metric recorded the
+// expected samples.
+func AssertHistogramSamples(t testing.TB, metric *prometheus.HistogramVec, samples ...float64) {
```

Comment:
@bduffany I found this helper easier to use than the existing one for my use case as I don't explicitly specify the buckets. What do you think?